src/components/CourseSetEdit.vue

```diff
@@ -12,70 +12,83 @@
+        Edit
+      </b-dropdown-item>
     </b-nav-item-dropdown>
 
     <b-modal id="courseSet-modal" title="Course Set Settings">
       <p>
         Course sets allow you to save and switch among multiple selections of
         courses/sections.
       </p>
-      <b-input-group>
-        <b-form-input
-          v-model="newCourseSetName"
-          :state="newCourseSetExists"
-          placeholder="Course Set Name"
-          aria-lable="Course Set Name"
-          trim
-          @keyup.enter="createNewCourseSet"
-        ></b-form-input>
-        <b-input-group-append>
-          <b-button
-            @click="createNewCourseSet"
-            style="
-              border-top-right-radius: 0.25rem;
-              border-bottom-right-radius: 0.25rem;
-            "
-            :disabled="!newCourseSetExists"
-            :title="newCourseSetExists ? '' : 'Disabled'"
-            >Add Course Set</b-button
-          ></b-input-group-append
+      <ul class="list-group list-group-flush">
+        <li class="list-group-item">
+          <h5 class="mb-0">Course Sets:</h5>
+        </li>
+        <li
+          class="list-group-item"
+          v-for="courseSet in Object.keys(getCourseSets)"
+          :key="courseSet"
         >
-        <b-form-invalid-feedback>
-          Must be a unique name
-        </b-form-invalid-feedback>
-        <!-- I dont actually show any form valid feedback, but having this here keeps
-           The page nicely spaced out and not bouncing-->
-        <b-form-valid-feedback id="valid-feedback">
-          Must be a unique name
-        </b-form-valid-feedback>
-      </b-input-group>
+          <font-awesome-icon
+            v-if="Object.keys(getCourseSets).length > 1"
+            :icon="['fas', 'trash']"
+            class="open_close_icon, trash-btn"
+            @click="removeCourseSet(courseSet)"
+          ></font-awesome-icon>
+          {{ courseSet }}
+        </li>
+        <li class="list-group-item">
+          <b-input-group>
+            <b-form-input
+              v-model="newCourseSetName"
+              :state="newCourseSetExists"
+              placeholder="Course Set Name"
+              aria-lable="Course Set Name"
+              trim
+              @keyup.enter="createNewCourseSet"
+            ></b-form-input>
+            <b-input-group-append>
+              <b-button
+                @click="createNewCourseSet"
+                style="
+                  border-top-right-radius: 0.25rem;
+                  border-bottom-right-radius: 0.25rem;
+                "
+                :disabled="!newCourseSetExists"
+                :class="{
+                  'btn-success': newCourseSetExists,
+                }"
+                :title="newCourseSetExists ? '' : 'Disabled'"
+                >Add Course Set</b-button
+              ></b-input-group-append
+            >
+            <b-form-invalid-feedback>
+              Must be a unique name
```

Comment:
Can we change this to be either `Must be a unique name` when the name conflicts or `Course sets can't have an empty name` (or something similar) when no input is given?

Reply:
I'll need to look into this some more since this depends on external libraries.

Reply:
You can make this reliant on a variable in the component, similar to how the Course Set name is rendered.